rllib/policy/dynamic_tf_policy.py

```diff
@@ -164,28 +163,32 @@ def __init__(self,
                 for s in self.model.get_initial_state()
             ]
 
-        model_out, self._state_out = self.model(self._input_dict,
-                                                self._state_in, self._seq_lens)
-
-        # Create the Exploration object to use for this Policy.
-        self.exploration = self._create_exploration(action_space, config)
         timestep = tf.placeholder(tf.int32, (), name="timestep")
 
-        # Setup custom action sampler.
-        if action_sampler_fn:
-            sampled_action, sampled_action_logp = action_sampler_fn(
-                self, self.model, self._input_dict, obs_space, action_space,
-                explore, config, timestep)
-        # Create a default action sampler.
+        if forward_fn:
```

Comment:
Custom forward_fn (only used in DQN and SAC) no longer need to deal with any exploration hooks!